client/deposits/overview/instant-deposit-button.js

```diff
@@ -29,34 +66,68 @@ const InstantDepositModal = ( { amount, fee, net, onClose, onSubmit, inProgress
 					currency.formatCurrency( net / 100 )
 				) }
 			</Button>
+			<Button isDefault onClick={ onClose }>
+				{ __( 'Cancel', 'woocommerce-payments' ) }
+			</Button>
 		</Modal>
-	)
-}
+	);
+};
+
+const InstantDepositSubmitNotice = ( { error } ) => {
+	let message = '';
+	// TODO: supply proper error notices.
+	switch ( error.code ) {
+		case 'balance_insufficient':
+			message = __(
+				'Balance is not sufficient for instant deposit.',
+				'woocommerce-payments'
+			);
+			break;
+
+		default:
+			message = __(
+				'There was an error, please try again.',
+				'woocommerce-payments'
+			);
+	}
+
+	return (
+		<Notice status="error" isDismissible={ false }>
+			<p>
+				<strong>{ message }</strong>
+			</p>
+		</Notice>
+	);
+};
 
 // TODO: Properly style :allthethings:
-const InstantDepositButton = ( { balance: { amount, fee, net, transaction_ids } } ) => {
+const InstantDepositButton = ( {
+	balance: { amount, fee, net, transaction_ids: transactionIds },
+} ) => {
 	const [ isModalOpen, setModalOpen ] = useState( false );
 	const [ inProgress, setInProgress ] = useState( false );
+	const [ hasError, setHasError ] = useState( false );
 
 	// TODO: Use wp.data
 	const submit = async () => {
 		try {
 			setInProgress( true );
+			setHasError( false );
 			await apiFetch( {
 				path: '/wc/v3/payments/deposits',
 				method: 'POST',
 				data: {
 					type: 'instant',
-					transaction_ids,
+					transactionIds,
 				},
 			} );
 			// TODO: Success notice? Full-reload the page so the new deposit appears?
 		} catch ( err ) {
 			// TODO: Real error management
-			alert( 'An error has occurred: ' + err );
+			setHasError( err );
 		} finally {
 			setInProgress( false );
-			setModalOpen( false );
+			//setModalOpen( false ); // Commented out due to it was closing modal on error.
```

Comment:
What do you think it's better? Showing the error notice in the modal, or closing the modal and showing the error notice in the page? In any case, commented-out code shouldn't be here at all.

Reply:
I think showing the error in the modal is better, since they might be able to fix something and then submit again without closing it. The success should be on the page with the modal closed.

Reply:
> they might be able to fix something and then submit again without closing it.

There's nothing to do in the modal (as you have seen). There's no amount selector or any option the merchant can check/uncheck. So I don't know if there's any point on telling the merchant to "just click the button again". Anyway, this is an extremely minor detail :)

Reply:
Ya, I figured we could get the functionality there, then get the actual messages in place after the fact.

From the support side of things, it would be awesome if we could determine if logging was on or not, then base messages off of that. If logging was not on, advise them where to turn it on, then tell them to try again. If it is on, advise them where to review the logs for the error. Something like that may save a support interaction, and if not, it would help the customer experience with support.

Reply:
>  it would be awesome if we could determine if logging was on or not, then base messages off of that.

This is a fantastic idea in general, not only for this particular error message. Something like a smarter `ErrorMessage` component that appends a `[See logs]` or `[Click here to enable logs]` link to the provided error message.

Do you mind creating an issue for that? It would be a great "maintenance" / "tech debt" task.

Reply:
Created the issue here: #1346 . All I could find for tagging purposes was _Enhancement_.